common/scala/src/main/scala/whisk/common/Config.scala

```diff
@@ -42,113 +42,120 @@ import scala.util.Try
+      .Map[String, String]() ++= requiredProperties
+    Config.readPropertiesFromEnvironment(required, env)
+
+    // for optional value, assign them a default from the required properties list
+    // to prevent loss of a default value on a required property that may not otherwise be defined
+    val optional = scala.collection.mutable
+      .Map[String, String]() ++= optionalProperties.map { k =>
+      k -> required.getOrElse(k, null)
     }
+    Config.readPropertiesFromEnvironment(optional, env)
+
+    required ++ optional
+  }
 }
 
 /**
  * Singleton object which provides global methods to manage configuration.
  */
 object Config {
-    /**
-     * Reads a Map of key-value pairs from the environment -- store them in the
-     * mutable properties object.
-     */
-    def readPropertiesFromEnvironment(properties: scala.collection.mutable.Map[String, String], env: Map[String, String])(implicit logging: Logging) = {
-        for (p <- properties.keys) {
-            val envp = p.replace('.', '_').toUpperCase
-            val envv = env.get(envp)
-            if (envv.isDefined) {
-                logging.info(this, s"environment set value for $p")
-                properties += p -> envv.get.trim
-            }
-        }
+
+  /**
+   * Reads a Map of key-value pairs from the environment -- store them in the
+   * mutable properties object.
+   */
+  def readPropertiesFromEnvironment(properties: scala.collection.mutable.Map[String, String],
+                                    env: Map[String, String])(implicit logging: Logging) = {
+    for (p <- properties.keys) {
+      val envp = p.replace('.', '_').toUpperCase
+      val envv = env.get(envp)
+      if (envv.isDefined) {
+        logging.info(this, s"environment set value for $p")
+        properties += p -> envv.get.trim
+      }
     }
+  }
 
-    /**
-     * Checks that the properties object defines all the required properties.
-     *
-     * @param required a key-value map where the keys are required properties
-     * @param properties a set of properties to check
-     */
-    def validateProperties(required: Map[String, String], properties: Map[String, String])(implicit logging: Logging): Boolean = {
-        required.keys.forall { key =>
-            val value = properties(key)
-            if (value == null) logging.error(this, s"required property $key still not set")
-            value != null
-        }
+  /**
+   * Checks that the properties object defines all the required properties.
+   *
+   * @param required a key-value map where the keys are required properties
+   * @param properties a set of properties to check
+   */
+  def validateProperties(required: Map[String, String], properties: Map[String, String])(
+      implicit logging: Logging): Boolean = {
+    required.keys.forall { key =>
+      val value = properties(key)
+      if (value == null)
+        logging.error(this, s"required property $key still not set")
+      value != null
     }
+  }
```

Comment:
FWIW I got a slightly different wrapping when trying the scalafmt plugin in intellij, using the same .scalafmt.conf:
```
  def validateProperties(required: Map[String, String], properties: Map[String, String])(
      implicit logging: Logging): Boolean = {
    required.keys.forall { key =>
      val value = properties(key)
      if (value == null) logging.error(this, s"required property $key still not set")
      value != null
    }
  }
```